### PR TITLE
Implement Code Snippets

### PR DESCRIPTION
**Pull request by strayacode**
- implements code snippets for bsts, but still improvements to be made ui wise
- fix bug where timeline slider value gets set to NaN when no animations get added to the timeline

### CODE REVIEW

File: client/src/components/CodeSnippet/CodeSnippet.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React, { FC, useEffect } from 'react';
+import { Box, Typography } from '@mui/material';
```
Comment (Tymotex) [inline client/src/components/CodeSnippet/CodeSnippet.tsx:2]: FYI: if you're using vscode, shift+alt+o removes unused imports and sorts them for you: https://stackoverflow.com/questions/46722701/is-there-a-way-to-remove-unused-imports-and-declarations-from-angular-2

Comment (strayacode) [reply client/src/components/CodeSnippet/CodeSnippet.tsx:2]: Oh cool. I fixed for CodeSnippet.tsx but how can i remove all unused imports for the whole project easily?

File: client/src/visualiser-src/binary-search-tree-visualiser/initialiser.ts [hunk 2]
```diff
@@ -33,6 +33,7 @@ const initialise = (): any => {
   const rotateRight = (val: number, updateSlider: (val: number) => void) => {
     controller.finish();
     const animationSequence: BSTAnimationProducer = bst.rotateRight(val);
+    console.log(animationSequence);
```
Comment (jlin2002) [inline client/src/visualiser-src/binary-search-tree-visualiser/initialiser.ts:36]: Get rid of this console log

**Review by jlin2002 — CHANGES_REQUESTED** [review]

**Review by jlin2002 — APPROVED** [review]

**Review by jlin2002 — CHANGES_REQUESTED** [review]
Console log was left behind

File: client/src/components/CodeSnippet/CodeSnippet.tsx (hunk 1 above)
Comment (Tymotex) [reply client/src/components/CodeSnippet/CodeSnippet.tsx]: Interesting, I can't find such a functionality for VSCode. I think this ends up being a responsibility for the linter: https://stackoverflow.com/questions/64365300/how-can-i-remove-unused-imports-declarations-from-the-entire-project-of-react-ty.

**Review by jlin2002 — APPROVED** [review]

**Pull request merged**